src/storage/all/mod.rs

```diff
@@ -28,7 +28,8 @@ use parking_lot::{lock_api::RawRwLock as _, RawRwLock};
 // we use a HashMap, it can reallocate, but even in this case the storages won't move since they are boxed
 pub struct AllStorages {
     lock: RawRwLock,
-    storages: UnsafeCell<HashMap<TypeId, Storage, BuildHasherDefault<TypeIdHasher>>>,
+    // TODO: Re-include the TypeIdHasher as a StorageIdHasher
+    storages: UnsafeCell<HashMap<StorageId, Storage /*, BuildHasherDefault<TypeIdHasher>*/>>,
```

Comment:
Do we need to re-implement our own hash now that we are using `StorageId` now, or is the default one going to be fine?

Reply:
The default one is (and always was) fine. It was a good opportunity to learn about `Hash` and `Hasher` but it's probably a good time to remove the custom hasher, it'll simplify things.